planner/core/stats.go

```diff
@@ -127,30 +139,33 @@ func (ds *DataSource) deriveStats() (*property.StatsInfo, error) {
 	return ds.stats, nil
 }
 
-func (p *LogicalSelection) deriveStats() (*property.StatsInfo, error) {
-	if p.stats != nil {
-		return p.stats, nil
-	}
-	childProfile, err := p.children[0].deriveStats()
-	if err != nil {
-		return nil, err
-	}
-	p.stats = childProfile.Scale(selectionFactor)
+// DeriveStats implement LogicalPlan DeriveStats interface.
+func (p *LogicalSelection) DeriveStats(childStats []*property.StatsInfo) (*property.StatsInfo, error) {
+	p.stats = childStats[0].Scale(selectionFactor)
 	return p.stats, nil
 }
 
-func (p *LogicalUnionAll) deriveStats() (*property.StatsInfo, error) {
+func (p *LogicalUnionAll) recursiveDeriveStats() (*property.StatsInfo, error) {
```

Comment:
Seems this code logic is the same with `baseLogicalPlan`'s? Except that `baseLogicalPlan` only handles one child, while `LogicalUnionAll` handles multi-child.